### PR TITLE
add dev vagrantfile

### PR DESCRIPTION
**Pull request by aweiteka**
Signed-off-by: Aaron Weitekamp <aweiteka@redhat.com>

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  # Provider-specific configuration so you can fine-tune various
+  # backing providers for Vagrant. These expose provider-specific options.
+  # Example for VirtualBox:
+  #
+  config.vm.provider "virtualbox" do |vb|
+  #   # Display the VirtualBox GUI when booting the machine
+  #   vb.gui = true
+  #
+  #   # Customize the amount of memory on the VM:
+    vb.memory = "2048"
```
Comment (mrunalp) [inline Vagrantfile]: I would suggest making this 4096 if someone intends to run kube local cluster as well.

Comment (TomSweeneyRedHat) [reply Vagrantfile]: Or maybe leave this as is and add a note about 4096 and kube local cluster?

Comment (aweiteka) [reply Vagrantfile]: I noticed 1GB is not enough to compile. 4GB is probably best so devs can be successful right away.

Comment (rhatdan) [reply Vagrantfile]: Yes lets make it large and then allow users to try to change it down.

**Comment by aweiteka** [comment]
@mrunalp update pushed to pre-config the dev environment. Maybe we can merge then see what else folks want to add?

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@mrunalp Should we merge or close this one?

**Comment by mrunalp** [comment]
This needs a rebase.

**Comment by rhatdan** [comment]
@aweiteka Could you rebase?

**Comment by aweiteka** [comment]
@rhatdan I updated this to reflect the current dev env and F27, rebased.

**Comment by rhatdan** [comment]
@mrunalp @runcom WDYT
No need to run e2e, I think?

**Pull request merged**